Signals/DatabaseMaintenanceTool/Program.cs

```diff
@@ -44,24 +44,20 @@ private static void CreateDatabaseIfNotExists()
             string databaseName = "Signals";
             string fileName = Path.Combine(AppDomain.CurrentDomain.GetData(DataDirectoryDataKey).ToString(), databaseName + ".mdf");
 
-            string connectionString = ConfigurationManager.ConnectionStrings["signals"].ConnectionString;
+            string connectionString = ConfigurationManager.ConnectionStrings["master"].ConnectionString;
 
-            if (!File.Exists(fileName))
+            using (var connection = new System.Data.SqlClient.SqlConnection(connectionString))
             {
-                using (var connection = new System.Data.SqlClient.SqlConnection(connectionString))
+                connection.Open();
+                using (var command = connection.CreateCommand())
                 {
-                    connection.Open();
-                    using (var command = connection.CreateCommand())
-                    {
-                        command.CommandText
-                            = string.Format("CREATE DATABASE {0} ON PRIMARY (NAME={0}, FILENAME='{1}')", databaseName, fileName);
-                        command.ExecuteNonQuery();
+                    command.CommandText
+                        = string.Format("IF EXISTS (SELECT name FROM master.dbo.sysdatabases WHERE name = '{0}') DROP DATABASE {0}", databaseName);
+                    command.ExecuteNonQuery();
 
```

Comment:
to be extra sure: after this line if file still exists - it should be manually removed (can happen if file gets detached for mssql)